examples/imagenet.py

```diff
@@ -0,0 +1,363 @@
+import os
+import warnings
+
+import hydra
+import torch
+import torch.nn as nn
+import torch.backends.cudnn as cudnn
+import torch.distributed as dist
+import torchvision.models as models
+import torchvision.datasets as datasets
+import torchvision.transforms as transforms
+from torch.optim import Adam
+
+best_acc1 = 0
+
+
+@hydra.main(config_name="imagenetconf")
+def main(cfg):
+    if cfg.seed is not None:
+        random.seed(cfg.seed)
+        torch.manual_seed(cfg.seed)
+        cudnn.deterministic = True
+        warnings.warn(
+            "You have chosen to seed training. "
+            "This will turn on the CUDNN deterministic setting, "
+            "which can slow down your training considerably! "
+            "You may see unexpected behavior when restarting "
+            "from checkpoints."
+        )
+    if cfg.gpu is not None:
+        warnings.warn(
+            "You have chosen a specific GPU. This will completely "
+            "disable data parallelism."
+        )
+    if cfg.dist_url == "env://" and cfg.world_size == -1:
+        cfg.world_size = int(os.environ["WORLD_SIZE"])
+
+    global best_acc1
+
+    if cfg.gpu is not None:
+        print("Use GPU: {} for training".format(cfg.gpu))
+
+    if cfg.distributed:
+        if cfg.dist_url == "env://" and cfg.rank == -1:
+            cfg.rank = int(os.environ["RANK"])
+        if cfg.multiprocessing_distributed:
+            cfg.rank = cfg.rank * cfg.ngpus_per_node + gpu
+        dist.init_process_group(
+            backend=cfg.dist_backend,
+            init_method=cfg.dist_url,
+            world_size=cfg.world_size,
+            rank=cfg.rank,
+        )
+    if cfg.pretrained:
+        print("=> using pre-trained model '{}'".format(cfg.arch))
+        model = models.__dict__[cfg.arch](pretrained=True)
+    else:
+        print("=> creating model '{}'".format(cfg.arch))
+        model = models.__dict__[cfg.arch]()
+
+    if not torch.cuda.is_available():
+        print("using CPU, this wil be slow")
+    elif cfg.distributed:
+        if cfg.gpu is not None:
+            torch.cuda.set_device(cfg.gpu)
+            model.cuda(cfg.gpu)
+            cfg.batch_size = int(cfg.batch_size / cfg.ngpus_per_node)
```

Comment:
Could we retain some of the original comments (from pytorch example). For example
```
# When using a single GPU per process and per
# DistributedDataParallel, we need to divide the batch size
 # ourselves based on the total number of GPUs we have
```

These comments make it easier to understand the why of these operations.